ExtLibs/ArduPilot/Mavlink/MAVLinkInterface.cs

```diff
@@ -1923,7 +1923,17 @@ public async Task<MAVLinkParamList> getParamListAsync(byte sysid, byte compid)
                                  (indexsreceived.Count * 100) / param_total);
                         if (frmProgressReporter != null)
                             this.frmProgressReporter.UpdateProgressAndStatus(
-                                (indexsreceived.Count * 100) / param_total, "Already Got param " + paramID);
+                                (indexsreceived.Count * 100) / param_total, "Already Got param " + "(" + indexsreceived.Count.ToString() + "/" + param_total.ToString() + ") " + paramID);
+                        return true;
+                    }
+                    // check if we future have it
+                    if (indexsreceived.Count < par.param_index)

```

Comment:
i dont understand what this is meant to be doing. you are forcing a perfect fetch, over lossy links. Mission planner already changes the fetch method based on how many it rx'ed and will switch to one by one if there is a high percentage of existing, vs lossed packets